kolibri/core/assets/src/api_resource.js

```diff
@@ -254,7 +411,6 @@ class Resource {
   getModel(id, data = {}) {
     let model;
     if (!this.models[id]) {
-      Object.assign(data, { [this.idKey]: id });
```

Comment:
I think getModel should let me know if the id I passed in cannot find any model. Return a new model kinda surprise me. I prefer to explicitly call createModel instead.